docs/architecture.rst

```diff
@@ -149,66 +155,85 @@ if any of the environment variables *SNF_IMAGE_DEV*,
 *SNF_IMAGE_PROPERTY_ROOT_PARTITION* or *SNF_IMAGE_TARGET* is unset or has a
 non-sane value.
 
+**InitializeDatasource**: If cloud-init is enabled, this task will create the
```

Comment:
"cloud-init is enabled" == SNF_IMAGE_PROPERTY_CLOUD_INIT is set, right? Maybe just state "This task will create the needed files..." here.